security/WFCORE-5466_Remove_Legacy_Vault_Integration.adoc

```diff
@@ -0,0 +1,312 @@
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE-5466[WFCORE-5466 - Remove Vault support entirely from WildFly Core]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFCORE-5485[WFCORE-5485 - Bump the CLI configuration schema to version 4]
+* https://issues.redhat.com/browse/WFLY-14912[WFLY-14912 - Remove Vault support entirely from WildFly]
+* https://issues.redhat.com/browse/EAP7-1745[EAP7-1745 - Remove Vault support]
+* https://issues.redhat.com/browse/WFLY-14943[WFLY-14943 - Once all legacy security removal is in perform a full refactor]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [x] QE
+
+As a removal RFE this does not require new feature test development, however both teams will need
+to review existing testsuites for tests dependent on the Vault integration.  Identified tests will
+either need deleting or converting to a credential store equivalent.
+
+=== Affected Projects or Components
+
+* WildFly Core
+* WildFly
+
+=== Other Interested Projects
+
+* [x] External Testsuites
+
+* [ ] Quickstarts
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+As a removal task, this section is describing the steps being undertaken rather than a set of
+"requirements".
+
+=== Hard Requirements
+
+The command line vault utility scripts `vault.bat`, `vault.ps1`, and `vault.sh` are all removed
+from the server.
+
+The next version of the management model is updated to remove all vault resources.  Any use of
+vault resources in the latest version will result in a usual unexpected element error.
+
+Where configurations are read using older model versions if they contain a vault definition this
+will be logged as a warning and the vault configuration dropped at the next write.  This is to
+allow older configurations to still be handled in admin mode.
+
+Vault expressions are not prevented from occurring within the management model for two reasons:
+
+* The WildFly Elytron expression encryption can use a similar format.
+* Domains will need to handle configurations containing vault expressions for older host controllers.
+
+The vault elements in the WildFly CLI config are removed from the next version of the schema.  The
+presence of the element in ALL versions of configuration will result in a parsing error being
+reported.  The parsing is more abrubt in the CLI as this is only used for configuring SSL so will
+need immediate attention.
+
+All vault integration code is deleted from WildFly Core reducing the dependence on PicketBox.
+
+All vault test utilities are deleted and removed from the testsuite.
+
+* TODO - Describe impact on Galleon layers.
```

Comment:
I believe the relationship to Galleon (in general, not just layers) is:

1) There are ee-feature-pack and servlet-feature-pack 'vault' layers (latter no longer relevant to the main WF dist). But these provide no configuration; they only exist to allow users to ensure that the picketbox package and its deps are provisioned in contexts where they otherwise would not be. So these could be removed altogether (the right thing to do I think, but would break configs that for some reason directly specified these layers), or just turned essentially into no-ops.

2) There's a vaulttools package used to ensure vault related scripts are provisioned. This can be dropped.